ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/HD/BackingStore.hs

```diff
@@ -0,0 +1,354 @@
+{-# LANGUAGE DataKinds                  #-}
+{-# LANGUAGE DeriveAnyClass             #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE DerivingVia                #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE LambdaCase                 #-}
+{-# LANGUAGE Rank2Types                 #-}
+{-# LANGUAGE StandaloneDeriving         #-}
+
+-- | A store for key-value maps that can be extended with deltas.
+--
+-- Its intended use is for storing data structures from the 'LedgerState' and
+-- update them with differences produced by executing the Ledger rules.
+module Ouroboros.Consensus.Storage.LedgerDB.HD.BackingStore (
+    -- * Backing store interface
+    BackingStore (..)
+  , BackingStoreInitialiser (..)
+  , BackingStorePath (..)
+  , BackingStoreValueHandle (..)
+  , InitFrom (..)
+  , RangeQuery (..)
+  , bsRead
+  , initFromCopy
+  , initFromValues
+  , withBsValueHandle
+    -- * An in-memory backing store
+  , StoreDirIsIncompatible (..)
+  , TVarBackingStoreExn (..)
+  , newTVarBackingStoreInitialiser
+    -- * A trivial backing store
+  , trivialBackingStore
+    -- * Ledger DB wrappers
```

Comment:
Add this section header to the code below as well

Reply:
Done